### PR TITLE
REST API Field Controller: Simplify Array Recursion

### LINKED ISSUE #10649: PHP Warnings caused by new REST API Field Controller
#### Steps to reproduce the issue

I ran into the following issues on a fresh install running `master`, when going to Posts > Add New:

```
Warning:  array_fill(): Number of elements must be positive in /wp-content/plugins/jetpack-dev/_inc/lib/core-api/class-wpcom-rest-field-controller.php on line 308
Warning:  array_fill(): Number of elements must be positive in /wp-content/plugins/jetpack-dev/_inc/lib/core-api/class-wpcom-rest-field-controller.php on line 309
Warning:  array_map(): Argument #3 should be an array in wp-content/plugins/jetpack-dev/_inc/lib/core-api/class-wpcom-rest-field-controller.php on line 310
Warning:  array_combine() expects parameter 2 to be array, null given in wp-content/plugins/jetpack-dev/_inc/lib/core-api/class-wpcom-rest-field-controller.php on line 312
```

Related: #10600

### PR DESCRIPTION
**Pull request by mdawaffe**
Fixes #10649

#### Changes proposed in this Pull Request:
* Replace complicated `array_map()` with `foreach`.
* Fixes PHP Warning for PHP < 5.6 when there are no items in the array.

##### Automated Tests
* `phpunit --group publicize`
* `phpunit --group rest-api`

##### Manual Tests
1. Create a site using this branch, WordPress 4.9.8, Gutenberg, **PHP 5.4**
2. Connect Jetpack.
3. Go to wp-admin/ -> Posts -> New

Prior to this patch: see PHP Warnings.

After this patch: see no PHP Warnings.

#### Proposed changelog entry for your changes:

None required.

### CODE REVIEW

**Comment by mdawaffe** [comment]
Let's hold off on merging this for now. (But please feel free to test!)

I'd like to think about it a bit more sometime when it's not late at night :)

**Review by ockham — DISMISSED** [review]
Fixes the warnings as described. Code looks good, too. :ship: at will!

**Review by tyxla — DISMISSED** [review]
LGTM :shipit:

File: _inc/lib/core-api/class-wpcom-rest-field-controller.php
Comment (mdawaffe) [reply _inc/lib/core-api/class-wpcom-rest-field-controller.php]: 90aa75201

**Review by ockham — APPROVED** [review]
Re-approving after lint fix

**Comment by mdawaffe** [comment]
WordPress.com: D21010-code.

**Review by dereksmart — APPROVED** [review]
Looks good tests well!

**Comment by jeherve** [comment]
Cherry-picked to `branch-6.8` in e847297750ae764576a064cbebbf150cacf75459

**Pull request merged**
